ffi/src/test_ffi.rs

```diff
@@ -0,0 +1,102 @@
+//! Utility functions used for testing ffi code
+
+use std::{ops::Not, sync::Arc};
+
+use crate::{expressions::SharedExpression, handle::Handle};
+use delta_kernel::{
+    expressions::{ArrayData, BinaryOperator, Expression, Scalar, StructData},
+    schema::{ArrayType, DataType, StructField, StructType},
+};
+
+/// Constructs a kernel expression that is passed back as a SharedExpression handle. The expected
+/// output expression can be found in `ffi/tests/test_expression_visitor/expected.txt` as a printed
+/// tree.
+///
+/// # Safety
+/// The caller is responsible for freeing the retured memory, either by calling
+/// [`free_kernel_predicate`], or [`Handle::drop_handle`]
+#[no_mangle]
+pub unsafe extern "C" fn get_testing_kernel_expression() -> Handle<SharedExpression> {
+    use Expression as Expr;
+
+    let array_type = ArrayType::new(
+        DataType::Primitive(delta_kernel::schema::PrimitiveType::Short),
+        false,
+    );
+    let array_data = ArrayData::new(array_type.clone(), vec![Scalar::Short(5), Scalar::Short(0)]);
+
+    let nested_fields = vec![
+        StructField::new("a", DataType::INTEGER, false),
+        StructField::new("b", DataType::array(array_type), false),
```

Comment:
There's an `impl From<ArrayType> for DataType`, and `StructField::new` accepts `Into<DataType>`:
```suggestion
        StructField::new("b", array_type, false),
```